docs/quick-start/nginx-ingress/README.md

```diff
@@ -0,0 +1,131 @@
+
+
+## Step 2 - Deploy an Example Service (Echoserver)
+```bash
+$ kubectl apply -f echoserver/00-namespace.yaml
+$ kubectl apply -f echoserver/deployment.yaml
+$ kubectl apply -f echoserver/service.yaml
+```
+
+We will first create an Ingress resource that listens on plain HTTP.
+
+```bash
+$ kubectl apply -f echoserver/ingress-notls.yaml
+```
+
+Now make sure the echo service is reachable at the domain name you added above, e.g. http://echo.example.com.
+
+
+## Step 3 - Deploy Cert Manager
+```bash
+$ helm install --name cert-manager --namespace kube-system stable/cert-manager
+```
+
+## Step 4 - Configure Letsentrypt Issuer
+
+With cert manager running, we must tell it about a service it can use to issue certificates for us. In this case we will configure letsencrypt. NB: the following code sets up letsencrypt "production". This will issue valid certificates but has a strict rate limit. If you are having difficulty and failing to get certificates, [switch this for letsencrypt's "staging" issuer](https://github.com/jetstack/cert-manager/blob/master/docs/user-guides/acme-http-validation.md) while you debug, otherwise you will hit the rolling 7 day rate limit.
+
+Substitute your own email address in the definition below and apply to your cluster.
+
+```yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-v01.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: user@example.com
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-prod-key
+    # Enable the HTTP-01 challenge provider
+    http01: {}
+```
+
+## Step 5 - Deploy a TLS Ingress Resource for the Echoserver
+We will now update the Ingress resource to listen on HTTPS and use a TLS certificate stored in a named secret. Cert manager will notice this change to the Ingress resource and generate the certificate for us automatically.
+
+Edit `echoserver/ingress-tls.yaml` to replace `echo.example.com` with your DNS name. While the previous, non-TLS, Ingress resource definition was simply configured to accept requests with any `host` header, TLS Ingress resources need to know which SNI header values to match, and `cert-manager` uses the same field as the CN in the issued certificate.
+
+Note how this updated Ingress resource uses an annotation to tell `cert-manager` to use the issuer previously defined to get a TLS certificate for this Ingress resource. The mere presence of this annotation key tells `cert-manager` to generate a certificate for this Ingress. It is generated for the domain name specified in the `tls` part of the Ingress spec, and stored in the secret named there.
+
+```bash
+$ kubectl apply -f echoserver/ingress-tls.yaml
+```
+
+The echo server should now be available over TLS at e.g. https://echo.example.com.
```

Comment:
We should tell users how they can watch for progress whilst the certificate is being validated/issued, and only tell the user the echoserver is available over TLS once `kubectl describe certificate` says the cert has been issued.

This will help users in future debug, and also stop people complaining it doesn't work for them!

We should **also** have a note to say that in some cases, an ingress controller may not see the new Secret resource containing the signed certificate, and instructing them how to use `kubectl get secret` (piped through openssl or something) to verify that the secret does in fact contain the correct data. & then probably also advise them to open an issue on the repo for their ingress controller too to get help.